workers/unity/Packages/com.improbable.gdk.core/Worker/WorkerConnector.cs

```diff
@@ -196,6 +215,89 @@ protected virtual string SelectDeploymentName(DeploymentList deployments)
             return null;
         }
 
+        /// <summary>
+        ///     Selects which login token to use to connect via the anonymous authentication flow.
+        /// </summary>
+        /// <param name="loginTokens">A list of available login tokens.</param>
+        /// <returns>The selected login token.</returns>
+        protected virtual string SelectLoginToken(List<LoginTokenDetails> loginTokens)
+        {
+            if (loginTokens.Count == 0)
+            {
+                throw new AuthenticationFailedException("Did not receive any login tokens. Do you have a valid deployment running?");
+            }
+
+            return loginTokens[0].LoginToken;
+        }
+
+        /// <summary>
+        ///     Retrieves the player identity token needed to generate a login token when using
+        ///     the anonymous authentication flow.
+        /// </summary>
+        /// <param name="authToken">The authentication token that you generated.</param>
+        /// <param name="playerId">The id of the player that wants to connect.</param>
+        /// <param name="displayName">The display name of the player that wants to connect.</param>
+        /// <returns>The player identity token.</returns>
+        protected virtual string GetDevelopmentPlayerIdentityToken(string authToken, string playerId, string displayName)
+        {
+            var result = DevelopmentAuthentication.CreateDevelopmentPlayerIdentityTokenAsync(
+                RuntimeConfigDefaults.LocatorHost,
+                RuntimeConfigDefaults.AnonymousAuthenticationPort,
+                new PlayerIdentityTokenRequest
+                {
+                    DevelopmentAuthenticationTokenId = authToken,
+                    PlayerId = playerId,
+                    DisplayName = displayName,
+                }
+            ).Get();
+
+            if (!result.HasValue)
+            {
+                throw new AuthenticationFailedException("Did not receive a player identity token.");
+            }
+
+            if (!string.IsNullOrEmpty(result.Value.Error))
```

Comment:
I think you should only check for `null` as the underlying C API returns a null pointer if there is no error. Not sure how this gets transformed into the C# struct. cc @davedissian

Reply:
Checked with David, we can actually just use the status code